egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,470 @@
+                        speech_dur,  # duration of the recording
+                        max_noises_recording  # Maximum number of point-source noises that can be added
+                        ):
+    if len(point_noise_list) > 0 and random.random() < pointsource_noise_addition_probability:
+        for k in range(random.randint(1, max_noises_recording)):
+            # pick the RIR to reverberate the point-source noise
+            noise = PickItemWithProbability(point_noise_list)
+            noise_rir = PickItemWithProbability(room.rir_list)
+            if noise.bg_fg_type == "background":
+                noise_addition_descriptor['noise_io'].append("wav-reverberate --duration={2} --impulse-response={1} {0} - |".format(noise.noise_file_location, noise_rir.rir_file_location, speech_dur))
+                noise_addition_descriptor['start_times'].append(0)
+                noise_addition_descriptor['snrs'].append(background_snrs.next())
+            else:
+                noise_addition_descriptor['noise_io'].append("wav-reverberate --impulse-response={1} {0} - |".format(noise.noise_file_location, noise_rir.rir_file_location))
+                noise_addition_descriptor['start_times'].append(round(random.random() * speech_dur, 2))
+                noise_addition_descriptor['snrs'].append(foreground_snrs.next())
+
+    return noise_addition_descriptor
+
+
+def GenerateReverberationOpts(room_dict,  # the room dictionary, please refer to MakeRoomDict() for the format
+                            point_noise_list, # the point source noise list
+                            iso_noise_list, # the isotropic noise list
+                            foreground_snrs, # the SNR for adding the foreground noises
+                            background_snrs, # the SNR for adding the background noises
+                            speech_rvb_probability, # Probability of reverberating a speech signal
+                            isotropic_noise_addition_probability, # Probability of adding isotropic noises
+                            pointsource_noise_addition_probability, # Probability of adding point-source noises
+                            speech_dur,  # duration of the recording
+                            max_noises_recording  # Maximum number of point-source noises that can be added
+                            ):
+    reverberate_opts = ""
+    noise_addition_descriptor = {'noise_io': [],
+                                 'start_times': [],
+                                 'snrs': []}
+    # Randomly select the room
+    room = PickItemWithProbability(room_dict)
+    # Randomly select the RIR in the room
+    speech_rir = PickItemWithProbability(room.rir_list)
+    if random.random() < speech_rvb_probability:
+        # pick the RIR to reverberate the speech
+        reverberate_opts += "--impulse-response={0} ".format(speech_rir.rir_file_location)
+
+    rir_iso_noise_list = FilterIsotropicNoiseList(iso_noise_list, speech_rir.rir_id)
```

Comment:
Do you want to call FilterIsotropicNoiseList for each recording ? It might be better to prepare a dictionary of istropic noises indexed by rir_id and just use it across calls to method.